cmd/zstream/zstream_recompress.c

```diff
@@ -77,7 +77,7 @@ zstream_do_recompress(int argc, char *argv[])
 	while ((c = getopt(argc, argv, "l:")) != -1) {
 		switch (c) {
 		case 'l':
-			if (sscanf(optarg, "%d", &level) != 0) {
+			if (sscanf(optarg, "%d", &level) == EOF) {
```

Comment:
0 would still be an undesirable result though right?  Or TBH anything except 1?

Reply:
I didn't bother looking up whether EOF is defined to be anything variable on any platforms.

Reply:
Right.  Shouldn't this really be updated to `!= 1` as @adamdmoss mentioned.  If we match anything other than a single field that's an error.

```
if (sscanf(optarg, "%d", &level) != 1) {
```

Reply:
This PR is why people don't contribute.

Reply:
> This PR is why people don't contribute.

I don't really understand if this is a joke or if I was being overpedantic.  In the latter case, sorry!

Reply:
It was a remark that the friction on fixing a one line bug is way too high, and I have a long list of code I will probably not bother trying to merge because nobody cares, they just make empty promises about "well that was bad, we should do better next time", don't change anything, then wonder why the number of developers and users keeps going down.